src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -382,6 +382,40 @@ uct_dc_mlx5_iface_dci_can_alloc(uct_dc_mlx5_iface_t *iface, uint8_t pool_index)
     return ucs_likely(pool->stack_top < iface->tx.ndci);
 }
 
+static UCS_F_ALWAYS_INLINE void
+uct_dc_mlx5_iface_dcis_array_copy(void *dst, void *src, size_t length)
+{
+    int i;
+    uct_dc_dci_t *src_dcis = (uct_dc_dci_t*)src;
+    uct_dc_dci_t *dst_dcis = (uct_dc_dci_t*)dst;
+
+    memcpy(dst_dcis, src_dcis, sizeof(uct_dc_dci_t) * length);
+
+    /* txqp is a queue and need to splice tail */
+    for (i = 0; i < length; ++i) {
+        ucs_queue_head_init(&dst_dcis[i].txqp.outstanding);
+        ucs_queue_splice(&dst_dcis[i].txqp.outstanding,
+                         &src_dcis[i].txqp.outstanding);
+    }
+}
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+uct_dc_mlx5_iface_dci_append(uct_dc_mlx5_iface_t *iface)
+{
+    uct_dc_dci_t empty_dci = {{{0}}};
+    void *old_buffer_p     = NULL;
+    size_t old_length      = ucs_array_length(&iface->tx.dcis);
+
+    *ucs_array_append_safe(&iface->tx.dcis, &old_buffer_p,
+                           return UCS_ERR_NO_MEMORY) = empty_dci;
+
+    if (old_buffer_p) {
+        uct_dc_mlx5_iface_dcis_array_copy(iface->tx.dcis.buffer, old_buffer_p,
+                                          old_length);
+        ucs_array_buffer_free(old_buffer_p);
+    }
+    return UCS_OK;
+}
```

Comment:
maybe not add this function, it seems like a code duplication, and revert to calling uct_dc_mlx5_iface_resize_and_fill_dcis(iface, dci_index + 1);